addons/crm/static/src/js/forecast/fill_temporal_service.js

```diff
@@ -0,0 +1,258 @@
+     *                         or until the last group with at least one record [forceFillingTo === false]
+     * @returns {Object} modified context
+     */
+    getContext(context, forceFillingTo=undefined) {
+        if (typeof forceFillingTo === 'undefined') {
+            forceFillingTo = !!this.end;
+        }
+        context = {
+            ...context,
+            fill_temporal: {
+                fill_from: this.start.format(FIELD_TYPE_TABLE[this.field.type].format),
+                min_groups: this.minGroups
+            }
+        };
+        if (forceFillingTo) {
+            context.fill_temporal.fill_to = moment(this.end)
+                                            .subtract(1, FIELD_TYPE_TABLE[this.field.type].minGranularity)
+                                            .format(FIELD_TYPE_TABLE[this.field.type].format);
+        }
+        return context;
+    }
+    /**
+     * @param {integer} minGroups minimum amount of groups to display, regardless of other constraints
+     */
+    setminGroups(minGroups) {
+        this.minGroups = minGroups || 1;
+    }
+}
+
+/**
+ * fill_temporal Service
+ *
+ * This service will be used to generate or retrieve fill_temporal periods
+ * 
+ * A specific fill_temporal period configuration will always refer to the same instance
+ * unless forceRecompute is true
+ */
+ var FillTemporalService = AbstractService.extend({
+    /**
+     * @override
+     */
+    start() {
+        this._super.apply(...arguments);
+        this._fillTemporalPeriods = {};
+    },
+    /**
+     * Get a fill_temporal period according to the configuration.
+     * The default initial fill_temporal period is the number of [granularity] from [start]
+     * to the end of the [cycle] reached after adding [minGroups] 
+     * i.e. we are in october 2020 :
+     *      [start] = 2020-10-01
+     *      [granularity] = 'month',
+     *      [cycle] = 12 (one year)
+     *      [minGroups] = 4,
+     *      => fillTemporalPeriod = 15 months (until end of december 2021)
+     * Once created, a fill_temporal period for a specific configuration will be stored
+     * until requested again. This allows to manipulate the period and store the changes
+     * to it. This also allows to keep the configuration when switching to another view
+     * 
+     * @param {Object} configuration
+     * @param {string} [modelName] directly taken from model.loadParams.modelName.
+     *                             this is the `res_model` from the action (i.e. `crm.lead`)
+     * @param {Object} [field] a dictionary with keys "name" and "type".
+     *                         name: Name of the field on which the fill_temporal should apply
+     *                               (i.e. 'date_deadline')
+     *                         type: 'date' or 'datetime'
+     * @param {string} [granularity] can either be : hour, day, week, month, quarter, year
+     * @param {integer} [minGroups] optional minimal amount of desired groups
```

Comment:
You can specify default arguments in JS doc like this:
```suggestion
     * @param {integer} [minGroups=4] optional minimal amount of desired groups
```